fastn/src/lib.rs

```diff
@@ -0,0 +1,505 @@
+mod tauri;
+
+/// Launch a native window displaying the `package_slug` got from cli args
+pub async fn fastn_ui_cli(matches: &clap::ArgMatches) -> fastn_core::Result<()> {
+    // We return with Ok so that other subcmds can run if `package_slug` is not provided
+    let Some(slug) = matches.get_one::<String>("package_slug") else {
+        return Ok(());
+    };
+
+    fastn_ui(slug).await
+}
+
+/// The main entry point for the fastn UI application.
+#[allow(unexpected_cfgs)]
+#[cfg_attr(mobile, tauri::mobile_entry_point)]
+pub async fn fastn_app() {
+    // Launch UI with the default package
+    // TODO: use lets-os for the default pkg
+    let slug = "design-sytem";
+    fastn_ui(slug).await.expect("Failed to launch fastn UI");
+}
+
+#[tracing::instrument]
+pub async fn fastn_ui(slug: &str) -> fastn_core::Result<()> {
+    let pkg_dir = {
+        let mut data_dir = dirs::data_dir().unwrap();
```

Comment:
Need some note on this. Where are these packages getting cloned? Can I run more than once instance of a package, eg checkout them twice, as each checkout will have its sqlite file for example. And if our file upload feature uploads files in the same folder, the two instances will have different files also going forward. Our editor may also modify these files etc. 

Actually for every package we should have a template package that should be checked out, as that is what most people want to do.